src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -1484,45 +1446,48 @@ TxQ::doRPC(Application& app) const
     using std::to_string;
 
     auto const view = app.openLedger().current();
-    auto const metrics = getMetrics(*view);
+    if (!view)
```

Comment:
I'm not sure how I feel about this because the check is redundant: the impementation of `OpenLedger` guarantees that `current` can _never_ be invalid. Having an `if` implies that it's possible and something to check for. An `assert` would be better suited because it communicates the "this can't happen!" nature of the check. I would be interested in knowing what others think?

Also, why `BOOST_ASSERT` over `assert`?

No change needed at this point.

Reply:
I kept the check because I wasn't 100% sure that it was impossible. If that's a known invariant, then I'll remove the check. There's no particular reason I used `BOOST_ASSERT` over `assert` - It's just habit.

Reply:
Oh, I just realized this is included in b6, so I won't make the change here.